src/main/java/uk/gov/companieshouse/authcodenotification/service/AuthCodeNotificationService.java

```diff
@@ -12,14 +12,25 @@
 @Service
 public class AuthCodeNotificationService {
 
+    private final PrivateDataRetrievalService privateDataRetrievalService;
+
+    private final EmailService emailService;
+
     @Autowired
-    private PrivateDataRetrievalService privateDataRetrievalService;
+    public AuthCodeNotificationService(PrivateDataRetrievalService privateDataRetrievalService,
+                                       EmailService emailService) {
+        this.privateDataRetrievalService = privateDataRetrievalService;
+        this.emailService = emailService;
+    }
 
-    public void sendAuthCodeEmail(String requestId, String companyNumber) throws ServiceException {
+    public void sendAuthCodeEmail(String requestId, String authCode, String companyNumber) throws ServiceException {
         var dataMap = new DataMap.Builder().companyNumber(companyNumber).build();
         ApiLogger.infoContext(requestId, "Send auth code email invoked", dataMap.getLogMap());
 
-        getOverseasEntityEmail(requestId, companyNumber, dataMap.getLogMap());
+        String email = getOverseasEntityEmail(requestId, companyNumber, dataMap.getLogMap());
+
+        // TODO Lookup company name
```

Comment:
What sub-task is handling this. Need to create a JIRA ticket for it if not yet existing. I'd be tempted to set 'TO BE RETRIEVED' or something rather than an empty string, but up to you.

Reply:
I'll leave as email as it will be done in a day or two. Added subtask https://companieshouse.atlassian.net/browse/ROE-2216 to cover the Todo